src/main/webapp/app/exercises/shared/participation/participation.utils.ts

```diff
@@ -3,9 +3,9 @@ import { createBuildPlanUrl } from 'app/exercises/programming/shared/utils/progr
 import { ProfileInfo } from 'app/shared/layouts/profiles/profile-info.model';
 
 export const setBuildPlanUrlForProgrammingParticipations = (profileInfo: ProfileInfo, participations: ProgrammingExerciseStudentParticipation[], projectKey?: string) => {
-    for (let i = 0; i < participations.length; i++) {
-        if (projectKey && participations[i].buildPlanId) {
-            participations[i].buildPlanUrl = createBuildPlanUrl(profileInfo.buildPlanURLTemplate, projectKey, participations[i].buildPlanId!);
+    participations.forEach((participation) => {
+        if (projectKey && participation.buildPlanId) {
+            participation.buildPlanUrl = createBuildPlanUrl(profileInfo.buildPlanURLTemplate, projectKey, participation.buildPlanId);
         }
-    }
+    });
```

Comment:
```js
    if (!projectKey) {
        return;
    }
    participations.filter((participation) => participation.buildPlanId).forEach((participation) => {
            participation.buildPlanUrl = createBuildPlanUrl(profileInfo.buildPlanURLTemplate, projectKey, participation.buildPlanId);
    });
```

Reply:
thanks for the suggestion. I'll use it. Unfortunately, TypeScript does not understand that filtering will make sure participation.buildPlanId is defined, so we need an additional ! in the end